control/src/main/java/org/kaazing/k3po/control/internal/Control.java

```diff
@@ -230,6 +242,28 @@ private void writeCommand(AbortCommand abort) throws IOException, CharacterCodin
         textOut.flush();
     }
 
+    private void writeCommand(NotifyCommand notify) throws IOException, CharacterCodingException {
+        OutputStream bytesOut = connection.getOutputStream();
+        CharsetEncoder encoder = UTF_8.newEncoder();
+        Writer textOut = new OutputStreamWriter(bytesOut, encoder);
+
+        textOut.append("NOTIFY\n");
+        textOut.append(format("barrier:%s\n", notify.getBarrier()));
```

Comment:
The control protocol is of the form 
```
FrameType
header1:value
header2:value
content-lenth: 
content
```

where the general agreed upon approach is unrecognized headers can be ignored.  But the content needs to be agreed upon.  This was the one thing I reviewed with @jfallows and I think this is better